### PR TITLE
refactor: Replace RenderStreaming components

### PR DESCRIPTION
**Pull request by karasusan**
This change replaces the `RenderStreaming` component to improve extensibility and maintainability.

### Changes of `RenderStreaming` component
- **WebSocket** is default for `signalingServerType` property
- `HardwareEncoderSupport` is disabled on default
- Remove `arrayButtonClickEvent` property (moved the property to `WebBrowserInputChannelReceiver` component)
- Added `runOnAwake` property
- Added `handlers` property

image

## Added components

- WebBrowserInputChannelReceiver
- InputSystemChannelReceiver
- InputSystemChannelSender

### WebBrowserInputChannelReceiver
This component treats messages of user inputs from browsers via DataChannel.
image

### InputSystemChannelReceiver
This component treats messages of user inputs from Unity via DataChannel.
image

### InputSystemChannelSender
This component sends messages of user inputs to Unity via DataChannel.
image

### Other changes
- Remove TextMeshPro dependencies from sample scenes

### CODE REVIEW

File: com.unity.renderstreaming/Runtime/Scripts/WebCamStreamer.cs [hunk 1]
```diff
@@ -38,28 +37,11 @@ IEnumerator StartVideo()
...
-
-            if (m_startVideoCorutine != null)
-            {
-                StopCoroutine(m_startVideoCorutine);
-            }
-
-            if (m_webCamTexture != null)
-            {
-                m_webCamTexture.Stop();
-            }
```
Comment (kannan-xiao4) [inline com.unity.renderstreaming/Runtime/Scripts/WebCamStreamer.cs:62]: I think better to have these process.

**Review by kannan-xiao4 — APPROVED** [review]
basically,lgtm

File: com.unity.renderstreaming/Runtime/Scripts/WebCamStreamer.cs (hunk 1 above)
Comment (karasusan) [reply com.unity.renderstreaming/Runtime/Scripts/WebCamStreamer.cs:62]: OK, I will fix it

Comment (karasusan) [reply com.unity.renderstreaming/Runtime/Scripts/WebCamStreamer.cs:62]: fixed it

**Review by kannan-xiao4 — APPROVED** [review]

**Pull request merged**